Upgrade rails-settings-cached. Fix specs. Prepare for settings select…

merqlove opened the pull request:
Upgrade rails-settings-cached. 
Fix it: https://github.com/huacnlee/rails-settings-cached/pull/110

Fix specs. 
Prepare for settings selection by key with coercion. 
Upgrade test deps.

merqlove committed:
Upgrade rails-settings-cached. Fix specs. Prepare for settings select by key with same coercion. Upgrade test deps

artofhuman commented on spec/settings_spec.rb:
```diff
@@ -118,26 +139,53 @@
+    #     fill_some_setting_check
+    #   end
+    # end
+
```
pr has many commented code, work in progress?

artofhuman commented on activeadmin_settings_cached.gemspec:
```diff
@@ -19,7 +19,7 @@ Gem::Specification.new do |s|
 
   s.add_dependency 'activeadmin'
-  s.add_dependency 'rails-settings-cached', '>= 0.5.3', '< 0.5.5'
+  s.add_dependency 'rails-settings-cached', '>= 0.6.0', '< 0.6.5'
```
can we save capability with old versions?

merqlove commented:
Yeah. I have upgraded rsc everywhere, but you can try.
Added checking for old defaults.
Within specs works well.

Now seems that we can update hash fields with merge.
It works when we select key in options.
Saving of plain hashes disallowed.
- Rollback rails-settings-cached v0.5.3 support.
- Add specs for model instance.
- Ability to save within namespaced keys.
- Add field_name helpers.
- Add save method for model instance.
- Added initial docs for keys.

merqlove commented:
Seems that something failed in CI. i will fix

merqlove committed:
Rollback rails-settings-cached v0.5.3 support.
Add specs for model instance.
Add field_name helpers.
Add save method for model instance.
Ability to save within namespaced keys.
Added initial docs for keys.

merqlove committed:
Fix keyed storage in model.
Update specs.

merqlove committed:
Fix AA error, `has been removed from the module tree but is still active`...

artofhuman commented on spec/model_spec.rb:
```diff
@@ -9,6 +9,164 @@
+      key: nil,
+      display: {'base.first_setting' => 'string', 'base.second_setting' => 'boolean'}
+    }
+  }
```
plz use `do end` for multiline blocks instead `{}`

artofhuman commented on Gemfile:
```diff
@@ -3,6 +3,7 @@ source 'https://rubygems.org'
 group :test do
   #gem 'turbolinks'
   gem 'activeadmin', github: 'activeadmin/activeadmin', branch: 'ab1d8d1a4a1af748ae7009d9d725193d780d0059'
+  gem 'rails-settings-cached', github: 'merqlove/rails-settings-cached', branch: 'master'
```
without merge fix https://github.com/huacnlee/rails-settings-cached/pull/110  in original repo we can`t merge this pr

merqlove replied:
@huacnlee won't merge PR with fix for get_all('base.') defaults...

merqlove replied:
It's merged 👍

artofhuman commented on activeadmin_settings_cached.gemspec:
```diff
@@ -19,7 +19,7 @@ Gem::Specification.new do |s|
 
   s.add_dependency 'activeadmin'
-  s.add_dependency 'rails-settings-cached', '>= 0.5.3', '< 0.5.5'
+  s.add_dependency 'rails-settings-cached', '>= 0.5.3', '< 0.6.5'
```
👍

artofhuman commented on spec/settings_spec.rb:
```diff
@@ -2,28 +2,48 @@
       end
       add_setting_resource
       add_second_setting_resource
+      # add_some_setting_resource
```
it`s needed ?

artofhuman commented:
Thanks for pr 👍

merqlove committed:
Cleanup.

merqlove committed:
Cleanup.
Restore use RSC from rubygems.

merqlove committed:
Drop ruby 2.0 from CI.
